types/firestore.d.ts

```diff
@@ -474,8 +474,9 @@ declare namespace FirebaseFirestore {
      * @param documentRef A reference to the document to be
      * created.
      * @param data The object to serialize as the document.
-     * @returns A promise that resolves with the result
-     * of the write. Throws an error if the write fails.
+     * @returns {Promise<WriteResult>} A promise that resolves with the result
```

Comment:
No need to include the return type in TypeScript. You should keep the type of the rejected result, since that is not apparent here.